src/vos/vos_obj.c

```diff
@@ -713,7 +716,7 @@ vos_obj_delete_internal(daos_handle_t coh, daos_unit_oid_t oid, bool only_delete
 	rc = umem_tx_end(umm, rc);
 
 out:
-	vos_obj_release(occ, obj, 0, true);
+	vos_obj_release(obj, 0, rc == 0);
```

Comment:
hmm, I don't see the value of reset the zombie flag.

Reply:
If we neither reset the **obj->obj_zombie** flag nor evict the object, then such zombie object will be kept in cache, and then the subsequent **vos_obj_hold()** will hit **-DER_AGAIN** repeatedly?

Reply:
Right, error will be returned when vos_obj_hold() is called against a zombie object. When zombie flag is set and "rc != 0", it indicates a local tx commit failure, which is a fatal error, I don't think we need to reset the zombie flag to pretend that everything is good. :)

Reply:
I mean that the cached zombie object will always fail subsequent vos_obj_hold(), as to nobody can operate on such object any longer?

Reply:
Yes. When zombie flag is set but rc != 0, it indicates a local tx failure, the whole system is likely out of order, we don't need bother to reset the zombie flag and pretend that everything is ok.

Reply:
Since then, it may be better to generate some error or warning message to draw the attention, that will be helpful for potential subsequent failed operation to understand what happened.